src/AcceptanceTests/ConfigureAzureStoragePersistence.cs

```diff
@@ -14,8 +14,9 @@ public class ConfigureEndpointAzureStoragePersistence : IConfigureEndpointTestEx
     public Task Configure(string endpointName, EndpointConfiguration configuration, RunSettings settings, PublisherMetadata publisherMetadata)
     {
         configuration.UsePersistence<AzureStoragePersistence, StorageType.Subscriptions>().ConnectionString(ConnectionString);
-        configuration.UsePersistence<AzureStoragePersistence, StorageType.Sagas>().ConnectionString(ConnectionString);
         configuration.UsePersistence<AzureStoragePersistence, StorageType.Timeouts>().ConnectionString(ConnectionString);
+        var sagaPersistence = configuration.UsePersistence<AzureStoragePersistence, StorageType.Sagas>().ConnectionString(ConnectionString);
+        sagaPersistence.AssumeSecondaryIndicesExist(true);
```

Comment:
I don't think the new overload with the bool is meaningful. Consider the following scenario. When a user already had `AssumeSecondaryIndicesExist()` that user already expressed the intent clearly in the code. Now we are forcing these users as well to change their code from `AssumeSecondaryIndicesExist()` to `AssumeSecondaryIndicesExist(true)`. Would it be better to leave it for users who already have opted in and to have an additional method for all the other users something like

- `AssumeSecondaryIndicesExist()`
- `IAmAwareThatNoSecondaryIndicesExistsAndIWantYouToContinueWithFullTableScanUntilITellYouToDoOtherwise()`

Reply:
I do not want to have an additional method to express something that is a single (mandatory) question and the answer should be either "yes" or "no". Yes, we did release in the current major a method that would only apply to the positive answer. Considering this is targeting vNext, I'd rather see that method deprecated and a proper overload to show up. I.e., I'm choosing slim API  over bloated one.

Reply:
Ok then how about we do the following

```
sagaPersistence.SecondaryIndexes(IndexCreation.AssumeExists|IndexCreation.IgnoreAllowTablescan)
```

or something the like. Basically using an enum instead of a bool. Credits go to @bording

Reply:
What value will it provide over `AssumeSecondaryIndicesExist(bool)`?
I can see it helpful when there are more scenarios that can take place. Is there another scenario that would require an additional flag?

Reply:
It expresses the intent much clearer than a bool. Especially because we enforce the usage of the method. When you look at 

```
AssumeSecondaryIndicesExist(false)
``` 

it is unclear what the implications of false are and why you can't just leave it out. By using an enum we can throw an exception saying that you have to make a choice. Also if we have to evolve it at some point into a third state we could. With the bool we can't

Reply:
> It expresses the intent much clearer than a bool. Especially because we enforce the usage of the method.

This argument doesn't hold since the method with boolean flag is also mandatory.

> Also if we have to evolve it at some point into a third state we could. With the bool we can't

If a third scenario is required, then we haven't thought it through and should start with API change until we know what's required. The problem we're facing is that full table scans should take place, or should not. No more, no less. I'm failing to identify a 3rd option here.

> it is unclear what the implications of false are and why you can't just leave it out.

```c#
AssumeSecondaryIndicesExist(false)
```
With proper documentation will provide the necessary information to make the decision.

> By using an enum we can throw an exception saying that you have to make a choice.

If `IndexCreation.AssumeExists|IndexCreation.IgnoreAllowTablescan` is blindly applied and exception is thrown, users would _still_ need to read the doco. 

So how is it better?

Reply:
How about 2 methods:
- sagaPersistence.SecondaryIndexesAreMigrated
- sagaPersistence.SecondaryIndexesAreNotYetMigrated

Reply:
It was already suggested [here](https://github.com/Particular/NServiceBus.Persistence.AzureStorage/pull/165#discussion_r138803168). It's far from ideal and it's confusing, just like showed with [`EnableInstallers`/`DisableInstallers`](https://github.com/Particular/PlatformDevelopment/issues/1417#issuecomment-329447923).

Reply:
Sorry, I was too lazy to read everything. It's different from the installers though, as in this case it is mandatory to choose one or the other, with installers that is not the case, there it depends on how the process was started.

Reply:
I'm going to park this discussion until @Particular/azure-maintainers call.
In my discussion with @bording the enum vs bool we went over the reason the API would need to exist and there is an aspect that as maintainers we've discussed in the past: being able to identify if data is migrated or not. That is shadowing the whole bool/not bool thread.

Previously, we have logged a warning if `AssumeSecondaryIndicesExist()` was not applied to help the new customers run more efficiently. Moving forward, we're saying we're mandating this information. While it helps us to optimize the code execution, it will not help existing customers moving forward. The problem will be still same problem - not able to identify if secondary indices exist for all sagas or not. Essentially, what that means is that this API will stick around, and will be there in the future. The only option to deprecate it would be to warn about deprecation and throw at run-time in the following versions, which is far from ideal.

I would like to discuss if we can how to
1. Remove the need for the new customers to use any API at all.
1. Provide a way for the existing customer migrate not only their code, but also their data. So that they don't need to guess if secondary indices exist for all of their sagas or not.